src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -594,5 +595,30 @@ internal static Location FirstOrNone(this ImmutableArray<Location> items)
         {
             return items.IsEmpty ? Location.None : items[0];
         }
+
+        internal sealed class SequenceEqualComparer<T> : IEqualityComparer<ImmutableArray<T>>
+        {
+            public static SequenceEqualComparer<T> Instance { get; } = new SequenceEqualComparer<T>();
+
+            public bool Equals(ImmutableArray<T> x, ImmutableArray<T> y)
```

Comment:
Why not `x.GetSpan().SequenceEquals(y.GetSpan())`

Reply:
We're running on Desktop, so how much slower is Slow Span?

Reply:
On second thought, it should still be worth it.

Reply:
I forgot the other problem. That helper requires T to implement IEquatable, which AnalyzerConfig.Section doesn't, and probably shouldn't. There's an open issue on CoreFX to add APIs for passing an IEqualityComparer https://github.com/dotnet/corefx/issues/35962